io/net/virt-net/veth_dlpar.py

```diff
@@ -56,99 +37,65 @@ def setUp(self):
         Gather necessary test inputs.
         '''
         self.interface = self.params.get('interface', default=None)
+        self.ipaddr = self.params.get("host_ip", default="")
+        self.netmask = self.params.get("netmask", default="")
         self.peer_ip = self.params.get('peer_ip', default=None)
         self.num_of_dlpar = int(self.params.get("num_of_dlpar", default='1'))
         self.vios_ip = self.params.get('vios_ip', '*', default=None)
         self.vios_user = self.params.get('vios_username', '*', default=None)
         self.vios_pwd = self.params.get('vios_pwd', '*', default=None)
-        self.login(self.vios_ip, self.vios_user, self.vios_pwd)
+        self.session = Session(self.vios_ip, user=self.vios_user,
+                               password=self.vios_pwd)
+        local = LocalHost()
+        self.networkinterface = NetworkInterface(self.interface, local)
+        try:
+            self.networkinterface.add_ipaddr(self.ipaddr, self.netmask)
+            self.networkinterface.save(self.ipaddr, self.netmask)
+        except Exception:
+            self.networkinterface.save(self.ipaddr, self.netmask)
+        self.networkinterface.bring_up()
```

Comment:
Please split this patch into 2 commits, one for IPAddr config, and one for vios command handling